### PR TITLE
Create testing framework

### PR DESCRIPTION
**Pull request by edelmanjm**
shhhh don't actually do anything with this yet, this is just here for "discussion"

### CODE REVIEW

File: test/build.gradle [hunk 1]
```diff
@@ -13,6 +13,7 @@ dependencies {
     compile navx()
     compile pathfinder()
     compile 'com.google.guava:guava:27.0-jre'
+    compile group: 'org.apache.commons', name: 'commons-math3', version: '3.6.1'
```
Comment (RobinsonZ) [inline test/build.gradle]: Use the other dependency notation plz
```Gradle
compile 'org.apache.commons:commons-math3:3.6.1'
```

**Comment by edelmanjm** [comment]
Alright should actually be ready to be merged now

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,3 +1,4 @@
+#Thu Oct 25 17:12:30 PDT 2018
```
Comment (RobinsonZ) [inline gradle/wrapper/gradle-wrapper.properties]: Revert this

**Comment by RobinsonZ** [comment]
(And as of #90 getting merged you need to change the root package, obviously.)

File: lib/src/main/java/org/team1540/rooster/testers/ResultWithMetadata.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
+package org.team1540.rooster.testers;
+
+import org.jetbrains.annotations.NotNull;
+
+public class ResultWithMetadata<R> {
```
Comment (RobinsonZ) [inline lib/src/main/java/org/team1540/rooster/testers/ResultWithMetadata.java]: Javadoc here?

**Pull request merged**